src/layouts/home-assistant.ts

```diff
@@ -64,6 +68,25 @@ export class HomeAssistantAppEl extends HassElement {
     }
   }
 
+  protected hassReconnected() {
+    super.hassReconnected();
+
+    // If backend has been upgraded, make sure we update frontend
+    if (this.hass!.connection.haVersion !== this._haVersion) {
+      if (supportsServiceWorker()) {
+        navigator.serviceWorker.getRegistration().then((registration) => {
```

Comment:
```suggestion
        navigator.serviceWorker.ready.then((registration) => {
```

Reply:
We don't need to know if the worker is active. We just want to know if we have one, and if we have one, we want to tell the browser that it should update the worker.